pkg/server/service.go

```diff
@@ -53,6 +58,16 @@ func Serve(ctx context.Context, pluginRegistry *plugins.Registry, additionalHand
 	return serveGatewayInsecure(ctx, pluginRegistry, serverConfig, authConfig.GetConfig(), storage.GetConfig(), additionalHandlers, adminScope)
 }
 
+func SetMetricKeys(appConfig *runtimeIfaces.ApplicationConfig) {
+	var keys = make([]contextutils.Key, 0, len(appConfig.MetricKeys))
+	for _, keyName := range appConfig.MetricKeys {
+		key := contextutils.Key(keyName)
+		keys = append(keys, key)
+	}
+	logger.Warningf(context.TODO(), "setting metrics keys to %+v", keys)
+	labeled.SetMetricKeys(keys...)
```

Comment:
Can you check if keys is empty and not call the function?

Reply:
done